test/fixtures/mson-inner-inheritance.json

```diff
@@ -20,33 +20,76 @@
               "content": {
                 "key": {
                   "element": "string",
-                  "content": "A"
+                  "content": "id"
                 },
                 "value": {
-                  "element": "object",
+                  "element": "number"
+                }
+              }
+            }
+          ]
+        },
+        {
+          "element": "object",
+          "meta": {
+            "id": "Person"
+          },
+          "content": [
+            {
+              "element": "member",
+              "content": {
+                "key": {
+                  "element": "string",
+                  "content": "user"
+                },
+                "value": {
+                  "element": "extend",
                   "content": [
                     {
-                      "element": "member",
-                      "content": {
-                        "key": {
-                          "element": "string",
-                          "content": "a"
-                        },
-                        "value": {
-                          "element": "string"
+                      "element": "object",
+                      "meta": {
+                        "ref": "User"
+                      },
+                      "content": [
+                        {
+                          "element": "member",
+                          "content": {
+                            "key": {
+                              "element": "string",
+                              "content": "id"
+                            },
+                            "value": {
+                              "element": "number"
+                            }
+                          }
                         }
-                      }
+                      ]
+                    },
+                    {
+                      "element": "object"
```

Comment:
Z wrote in comment above
:
> @klokane Careful the fixtures DOES NOT seem to be correct ￼
>  The elements in extend's content should be of the same type – in this case objects. For exampleof original members. It is not problem to remove if it is empty.

And if original is empty object, result is same :)

It is not problem if orginal emenet is empty, but there is question: which way is correct?